schemas/resolvers/resolver_mutations.ts

```diff
@@ -9,6 +11,17 @@ export const mutations_resolver = {
     authenticate(_: undefined, { input }: LoginUser) {
       return login(input);
     },
+    createProduct(
+      _: undefined,
+      { input }: CreateProduct,
+      context: ContextType
+    ) {
+      if (context.isAuth) {
```

Comment:
At this point you don't need to check `if(context.isAuth)` or `if(context.user)` since we are already throwing an error in our `@private` directive when user does not exist.